src/posts/post.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import Layout from '../layout';
+import { Center, Box, Heading, Code, Link } from '@chakra-ui/react'
+
+import { MDXProvider } from "@mdx-js/react"
+import SEO from "../components/seo";
+
+const components = {
+  h1: (props: string) => <Heading as='h1' size='xl' pb={10} {...props} />,
+  h2: (props: any) => <Heading as='h1' size='l' pb={8} {...props} />,
+  p: (props: any) => (
+    <p style={{ fontSize: "18px", lineHeight: 1.6 }} {...props} />
+  ),
+  a: (props: {href: string, children: string}) => (
+    <Link color='purple.400' href={props.href}>{props.children || 'Link'}</Link>
+  ),
+  code: (props: any) => (
+    <Center>
+        <Code bg="#3d3d3d" color="white" width={720} p={10} {...props} />
+    </Center>
+  )
+}
```

Comment:
This is maps MDX markdown tags to use Chakra, our UI framework. 

Write MDX and use markdown, MDX processes that input and generates HTML, here is where you can change how that markdown renders, so instead of generate an h1 tag, we generate a Chakra <Heading > tag.

This could be pulled out of this file and into it's own file for reuse.